core/controllers/base.py

```diff
@@ -362,12 +362,20 @@ def _render_exception_json_or_html(self, return_type, values):
         if return_type == feconf.HANDLER_TYPE_HTML and (
                 method == 'GET'):
             self.values.update(values)
-            if 'iframed' in self.values and self.values['iframed']:
-                self.render_template(
-                    'error-iframed.mainpage.html',
-                    iframe_restriction=None)
+            status = self.request.get('status', default_value=None)
+            # self.render_json(status)
+            if status:
+                if 'iframed' in self.values and self.values['iframed']:
+                    self.render_template(
+                        'error-iframed.mainpage.html',
+                        iframe_restriction=None)
+                else:
+                    self.render_template('error-page.mainpage.html')
             else:
-                self.render_template('error-page.mainpage.html')
+                redirect_uri = python_utils.convert_to_bytes(
+                    '%s?status=%s' % (self.request.uri, values['status_code']))
+                self.redirect(redirect_uri)
```

Comment:
If I understand this correctly, we are redirecting to the same handler if it is missing the `status` in the URL? I imagine that this can be somehow simplified to not include the redirect but to directly modify the URL?

Reply:
> If I understand this correctly, we are redirecting to the same handler if it is missing the status in the URL?

Yeah.

> I imagine that this can be somehow simplified to not include the redirect but to directly modify the URL?

I'm still checking for a better way to do this without having to redirect.